backend/src/handlers/category_operations.py

```
@@ -752,6 +801,188 @@ def delete_category_rule_handler(event: Dict[str, Any], user_id: str) -> Dict[st
+    except ValueError as e:
+        logger.error(f"Validation error extracting patterns: {str(e)}")
+        return create_response(400, {"message": str(e)})
+    except Exception as e:
+        logger.error(f"Error extracting patterns: {str(e)}")
+        return create_response(500, {"message": "Error extracting patterns"})
+
+def create_category_with_rule_handler(event: Dict[str, Any], user_id: str) -> Dict[str, Any]:
+    """Create category with pre-populated rule."""
+    try:
+        from services.pattern_extraction_service import PatternExtractionService
+        from models.category import MatchCondition
```

Flake8 (7.2.0)

[error] redefinition of unused 'MatchCondition' from line 12

(F811)